bench/src/jmh/java/org/pkl/core/Fibonacci.java

```diff
@@ -98,7 +100,23 @@ public long fib_lambda() {
 
 // kept similar to pkl code (class, instance method, long argument)
 class FibJavaImpl {
+  private Map<Long, Long> memo = new HashMap<>();
+
+  // Calculates the nth Fibonacci number using memoization to avoid redundant calculations.
   long fib(long n) {
-    return n < 2 ? n : fib(n - 1) + fib(n - 2);
+    // Base cases: 0 and 1
+    if (n < 2) {
+      return n;
+    }
+
+    // If the result for n is already calculated, return it from memo
+    if (memo.containsKey(n)) {
+      return memo.get(n);
+    }
+
+    // Otherwise, calculate Fibonacci recursively and store the result in memo
+    long result = fib(n - 1) + fib(n - 2);
+    memo.put(n, result);
+    return result;
```

Comment:
This class is intentionally written this way--we use this to compare Pkl's fibonacci performance to native Java (see comment on line 99)